sqlglot/dialects/snowflake.py

```diff
@@ -43,6 +43,7 @@
 
 
 DATE_PARTS = ["YEAR", "QUARTER", "MONTH", "WEEK", "DAY"]
+TIME_PARTS = ["HOUR", "MINUTE", "SECOND"]
```

Comment:
Why are we adding what is essentially the negation of `DATE_PARTS`? I thought we were going to mimic Brad's logic: `expression.text("unit").upper() not in DATE_PARTS`. Either that, or we should remove `DATE_PARTS` and do the `in TIME_PARTS` condition you introduced in this PR, but having both is redundant.